### ISSUE TITLE
constantmods json not found in subdirectory

### ISSUE THREAD

**Issue by offa** [priv-kweihmann/oelint-adv#777]
A constantmods json file configured in .oelint.cfg is not found if oelint is called from a subdirectory.

#### Project structure:

```
.oelint.cfg
.oelint-constants.json
recipes-dummy/dummy/dummy_0.1.bb
```

#### .oelint.cfg
```
[oelint]
[…]
constantmods =
    +.oelint-constants.json
```

#### .oelint-constants.json
```
{
    "variables": {
        "known": [
            "EXAMPLE"
        ]
    }
}
```

Called from project root everything works fine:

```bash
 oelint-adv recipes-dummy/dummy/dummy_0.1.bb
```

Now *cd* into subdirectory and pass `OELINT_CONFIG`:

```bash
export OELINT_CONFIG=$PWD/.oelint.cfg
cd recipes-dummy/dummy
oelint-adv dummy_0.1.bb
```

The config is picked up correctly, but the `.oelint-constants.json` isn't found as it seems not to account `OELINT_CONFIG` or the path relative to `oelint.cfg`:

```

Traceback (most recent call last):
  File "[…]/oelint_adv/core.py", line 447, in arguments_post
    with open(mod.lstrip('+-')) as _in:
         ~~~~^^^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '.oelint-constants.json'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "[…]/bin/oelint-adv", line 8, in <module>
    sys.exit(main())
             ~~~~^^
  File "[…]/oelint_adv/__main__.py", line 108, in main
    args = arguments_post(parse_arguments())
  File "[…]/oelint_adv/core.py", line 459, in arguments_post
    raise argparse.ArgumentTypeError(
        'mod file \'{file}\' is not a valid file'.format(file=mod))
```

- oelint-adv 8.1.4
- Linux / Python 3.13
